apps/dashboard/src/contract-ui/tabs/tokens/components/transfer-button.tsx

```diff
@@ -1,49 +1,160 @@
       <Button
         colorScheme="primary"
         leftIcon={<Icon as={FiSend} />}
         {...restButtonProps}
-        onClick={onOpen}
+        onClick={() => setOpen(true)}
         isDisabled={!hasBalance}
       >
         Transfer
       </Button>
+      <Sheet open={open} onOpenChange={setOpen}>
+        <SheetContent className="z-[10000]">
+          <SheetHeader>
+            <SheetTitle>Transfer tokens</SheetTitle>
+          </SheetHeader>
+          <form className="mt-10">
+            <div className="flex flex-col gap-6">
+              <FormControl isRequired isInvalid={!!form.formState.errors.to}>
+                <FormLabel>To Address</FormLabel>
+                <SolidityInput
+                  formContext={form}
+                  solidityType="address"
+                  placeholder={ZERO_ADDRESS}
+                  {...form.register("to")}
+                />
+                <FormHelperText>
+                  Enter the address to transfer to.
+                </FormHelperText>
+                <FormErrorMessage>
+                  {form.formState.errors.to?.message}
+                </FormErrorMessage>
+              </FormControl>
+              <FormControl
+                isRequired
+                isInvalid={!!form.formState.errors.amount}
+              >
+                <FormLabel>Amount</FormLabel>
+                <Input
+                  type="text"
+                  pattern={`^\\d+(\\.\\d{1,${decimalsQuery.data || 18}})?$`}
+                  {...form.register("amount")}
+                />
+                <FormHelperText>
+                  How many would you like to transfer?
+                </FormHelperText>
+                <FormErrorMessage>
+                  {form.formState.errors.amount?.message}
+                </FormErrorMessage>
+              </FormControl>
+            </div>
+          </form>
+          <SheetFooter className="mt-10">
+            <TransactionButton
+              transactionCount={1}
+              form={TRANSFER_FORM_ID}
+              isLoading={sendConfirmation.isPending}
+              type="submit"
+              colorScheme="primary"
+              isDisabled={!form.formState.isDirty}
+              onClick={form.handleSubmit((d) => {
+                trackEvent({
+                  category: "token",
+                  action: "transfer",
+                  label: "attempt",
+                });
+                const transaction = transfer({
+                  contract,
+                  amount: d.amount,
+                  to: d.to,
+                });
+                sendConfirmation.mutate(transaction, {
+                  onSuccess: () => {
+                    trackEvent({
+                      category: "token",
+                      action: "transfer",
+                      label: "success",
+                    });
+                    form.reset({ amount: "0", to: "" });
+                    onSuccess();
+                    setOpen(false);
```

Comment:
I guess u need to still be able to close it automatically when the mutation succeeds, that's fine u can use the `onOpenChange` prop on the sheet in combination with the trigger button and still have a state above

Reply:
ah okay